src/templates/finops-hub/modules/dataExplorer.bicep

```diff
@@ -0,0 +1,134 @@
+//==============================================================================
+// Parameters
+//==============================================================================
+
+@description('Required. Name of the hub. Used to ensure unique resource names.')
+param hubName string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Resource tags.')
+param tags object = {}
+
+@description('Required. The name of the Azure Key Vault instance.')
+param keyVaultName string
+
+//------------------------------------------------------------------------------
+// Variables
+//------------------------------------------------------------------------------
+
+// Generate globally unique cluster name: 4-22 chars; lowercase letters and numbers
+var safeHubName = replace(replace(toLower(hubName), '-', ''), '_', '')
+var clusterName = replace('${take(safeHubName, 22)}', '--', '-')
+
+//==============================================================================
+// Resources
+//==============================================================================
+
+resource cluster 'Microsoft.Kusto/clusters@2022-12-29' = {
+  name: clusterName
+  location: location
+  tags: tags
+  sku: {
+    capacity: 1
+    name: 'string'
+    tier: 'string'
+  }
+  identity: {
+    type: 'string'
+    userAssignedIdentities: {}
+  }
+  properties: {
+    acceptedAudiences: [
+      {
+        value: 'string'
+      }
+    ]
+    allowedFqdnList: [
+      'string'
+    ]
+    allowedIpRangeList: [
+      'string'
+    ]
+    enableAutoStop: true
+    enableDiskEncryption: false
+    enableDoubleEncryption: false
+    enablePurge: false
+    enableStreamingIngest: false
+    engineType: 'V3'
+    keyVaultProperties: {
+      keyName: keyVaultName
+      keyVaultUri: 'string'
+      keyVersion: 'string'
+      userIdentity: 'string'
+    }
+    languageExtensions: {
+      value: [
+        {
+          languageExtensionImageName: 'string'
+          languageExtensionName: 'string'
+        }
+      ]
+      value: [
+        {
+          languageExtensionImageName: 'string'
+          languageExtensionName: 'string'
+        }
+      ]
+    }
+    optimizedAutoscale: {
+      isEnabled: bool
+      maximum: int
+      minimum: int
+      version: int
+    }
+    publicIPType: 'string'
+    publicNetworkAccess: 'string'
+    restrictOutboundNetworkAccess: 'string'
+    trustedExternalTenants: [
+      {
+        value: 'string'
+      }
+    ]
+    virtualClusterGraduationProperties: 'string'
+    virtualNetworkConfiguration: {
+      dataManagementPublicIpId: 'string'
+      enginePublicIpId: 'string'
+      subnetId: 'string'
+    }
+  }
+  zones: [
+    'string'
+  ]
+}
+
+resource database 'Microsoft.Kusto/clusters/databases@2022-12-29' = {
+  name: 'Hub'
```

Comment:
Open to suggestions about the db name. I haven't put any thought into it yet...

Reply:
How about CostManagement?

Reply:
Long-term, the database will host more than just cost information. That's why I was starting with something more generic. Not sure if there are specific recommendations around when to split data into separate databases.

Reply:
Splitting tables across different databases will not affect performance. Having separate databases allows you to have better contol of permissions (since these are set per database, not per table), if you think it's valuable to you.